### DISCUSSION TITLE
Livewire v3: morphing more complex component tree: Block.get children(): currentNode can be null

### DISCUSSION THREAD

**Discussion by levu42** [alpinejs/alpine#3728] [7. Bugs (archived)]
I'm using Livewire 3 on a big codebase that I've upgraded a few betas ago to Livewire v3, and now I ran into a bug when updating a bit more complex component tree. I was unfortunately not able to make a reproducable test case as a bug report, but I found a JS fix inside alpine that fixes the issue:

In https://github.com/alpinejs/alpine/blob/5506b5c170699ae86b9d6e23f8c60f70bf780846/packages/morph/src/morph.js#L406 I run into the situation that `currentNode` is `null`, but not `undefined`.

Changing the condition from `currentNode !== undefined` to `currentNode !== undefined && currentNode !== null` fixed the problem for me.

I don't fully understand what's happening here, but I would love to have that fix upstream, so I don't need to bugfix my `livewire.esm.js` on every Livewire / alpine update ;)

**Comment by levu42** [comment]
PS: The error I was originally getting was that it couldn't read `nextSibling` on `null`, so that might be the error message other people might get.

**Comment by ekwoka** [comment]
Probably better to just do

```js
while (currentNode && currentNode !== this.endComment) {
```

This is definitely a place that typescript would have prevented this issue.

**Reply by levu42** [reply]
@ekwoka yes, nicer solution!

**Comment by levu42** [comment]
PR with a fix: https://github.com/alpinejs/alpine/pull/3729

**Reply by NickBourque** [reply]
This just cropped up for me as well - looking forward to seeing this fix getting merged!

**Reply by sawirricardo** [reply]
Any chance you have a fork for this to be fixed with livewire?

**Comment by emincer** [comment]
Seeing the same  Uncaught TypeError: Cannot read properties of null (reading 'nextSibling') down stream into LiveWIre 3 and Filament 3. Roadblocked, so hoping for a fix.

**Comment by joshhanley** [comment]
@levu42 the issue was actually to do with the morph markers creating mis-formed html when using the `@forelse/ @empty` directives. This has been fixed in PR https://github.com/livewire/livewire/pull/6522 and released in Livewire 3.0.1.

**Reply by ekwoka** [reply]
Seems valuable to fix the type safety issue regardless.

**Reply by joshhanley** [reply]
@ekwoka yeah fair

**Reply by levu42** [reply]
@joshhanley Thanks! Livewire 3.0.1 fixed it for me :)

**Reply by elchibek1** [reply]
Thanks man! You are awesome

**Comment by joydeep-bhowmik** [comment]
same problem still not fixed...
this issue is occurring with `@forelse/ @empty`  directive

**Reply by joshhanley** [reply]
@joydeep-bhowmik what patch version of Livewire are you using? Also can you provide some example code that demonstrates the issue so we can investigate? Thanks!

**Reply by NickBourque** [reply]
@joydeep-bhowmik Livewire `^3.0.1` fixed it for me. If you're still running `3.0.0` try upgrading.

**Reply by joydeep-bhowmik** [reply]
that might be the case .... can u please tell how to upgrade

**Reply by NickBourque** [reply]
`composer require livewire/livewire:^3.0.1`

**Reply by joydeep-bhowmik** [reply]
thanks
